GDevelop.js/scripts/generate-dts.mjs

```diff
@@ -0,0 +1,214 @@
+// @ts-check
+import { readFileSync, writeFileSync } from 'fs';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const bindingsFile = readFileSync(
+  __dirname + '/../Bindings/Bindings.idl',
+  'utf-8'
+);
+
+const PrimitiveTypes = new Map([
+  ['DOMString', 'string'],
+  ['long', 'number'],
+  ['double', 'number'],
+  ['void', 'void'],
+  ['boolean', 'boolean'],
+]);
+
+const none = Symbol('No default value');
+class Parser {
+  static i = 0;
+  /** @private */
+  static str = '';
+  /** @private */
+  static l = 0;
+
+  /** @param {string} str */
+  static set target(str) {
```

Comment:
I suggest:

```suggestion
  static setSource(str) {
```

(this is the source of your parser, not the target which would be the AST that the Parser emits)

I'm usually not a big fan of these magic get/set, they hide a function call and all you win is two characters (`.c` => `.c()`)

So when I read `.target = interfaceCode` below in the code, I was assuming the Parser context was not properly reset.